closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -344,7 +371,7 @@ JNIEXPORT jint JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_loadCrypto
     OSSL_get_error = (OSSL_get_error_t*)find_crypto_symbol(crypto_library, "ERR_get_error");
 
     /* Load Threading routines for OpenSSL 1.0.2 */
-    if (0 == ossl_ver) {
+    if ((OPENSSL_VERSION_1_0_0 <= ossl_ver) && (ossl_ver < OPENSSL_VERSION_1_1_0)) {
```

Comment:
If `ossl_ver < OPENSSL_VERSION_1_1_0` we've already verified that `OPENSSL_VERSION_1_0_0 <= ossl_ver` so only the first test is needed.

Reply:
Edited.